client/app.vue

```diff
@@ -1,6 +1,20 @@
 <script setup lang="ts">
 useTheme()
+useWebsocketAutoConnect()
 
+useEventBus(WebsocketPacketReceivedEvent).on(({ payload }) => {
+  switch (payload.type) {
+    case 'unknown':
+      console.log('Unknown packet received', payload) // eslint-disable-line no-console
+      break
+    case 'user-updated': {
+      // type-safe access to payload.data
+      const { username, email } = payload.data
+      console.log('User updated', username, email) // eslint-disable-line no-console
+      break
+    }
+  }
+})
```

Comment:
Nice work, just don't forget to remove these console.log or comment them. You could event implement auto-logout when a user gets disabled:

```suggestion
useEventBus(WebsocketPacketReceivedEvent).on(({ payload }) => {
  switch (payload.type) {
    case 'unknown':
      console.log('Unknown packet received', payload) // eslint-disable-line no-console
      break
    case 'user-updated': {
      if (!payload.data.enabled)
        logout()
        
      // type-safe access to payload.data
      const { username, email } = payload.data
      console.log('User updated', username, email) // eslint-disable-line no-console
      break
    }
  }
})
```

Reply:
I added logs just to show you how it's used and how it's type safe ; but the actual implementation is out of scope of this PR, so I'll leave the logs for now.